pkg/tagger/collectors/garden_main.go

```diff
@@ -31,10 +31,6 @@ type GardenCollector struct {
 
 // Detect tries to connect to the Garden API and the cluster agent
 func (c *GardenCollector) Detect(ctx context.Context, out chan<- []*TagInfo) (CollectionMode, error) {
-	if !config.IsFeaturePresent(config.CloudFoundry) {
-		return NoCollection, nil
-	}
```

Comment:
Any reason why this didn't work as expected? Should we rather fix `detectCloudFoundry()` ? (just curious about the bug, we can ship this fix as is in 7.30 and fix `detectCloudFoundry` in 7.31 ofc)

Reply:
This tests the connection to the socket of the garden process. But if it's not started yet at the time the agent starts, then we skip collection and there's no retry ever. The correct logic is in the `cloudfoundry.GetGardenUtil()` function just below this, that returns a retriable error.

Reply:
Can we expect the socket to exist even if `Garden` process is not started yet?

Reply:
Thanks for the explanation - in any case let's backlog something about this to fix `detectCloudFoundry` in 7.31 🙇

Reply:
Can we expect the socket to exist even if Garden process is not started yet?
It could be easily modified in `detectCloudFoundry`, otherwise it's useless and should be removed

Reply:
No the socket won't exist if the garden process is not started yet.
What's the use case for the `detectCloudFoundry` given we already have the `Detect` method ?

Reply:
> What's the use case for the detectCloudFoundry given we already have the Detect method ?

The goal was to avoid even trying to connect to garden if the env is not cloudfoundry

Reply:
Ah ! Then we should be able to check the config option `cloud_foundry` to check if we're even in a cloud foundry environment
http://github.com/Datadog/datadog-agent/blob/master/pkg/config/config.go#L590-L590

Reply:
cool thanks! a proper fix would be simple then 💯

Reply:
Should I make it for 7.30 instead of this quick and dirty one ?

Reply:
Ideally speaking we would want to auto set `cloud_foundry` based on some system/environment/files/whatever that the Agent could auto-detect (that's the purpose of `detectCloudFoundry`).
If not possible we can reverse it and use `cloud_foundry` as an indicator to activate the Cloud Foundry features.

I think it'd be better to fix it in 7.30.

Reply:
Well that flag in the config is the detector then. It's automatically set in the configuration when the agent is installed in a cloud foundry environment.
I'll make a PR